File: ZoneSentry/Models/PurchaseRequest.cs
namespace ZoneSentry.Models;

public class PurchaseRequest
{
    public int Id { get; set; }
    public int SellCost { get; set; }

    public Realty Realty { get; set; }
    public ApplicationUser NewOwner { get; set; }
}

public class PurchaseRequestDTO
{
    public int Id { get; set; }
    
    public int PaymentPerMonth { get; set; } 
    public int DurationInMonths { get; set; }
    
    public RealtyUserView Realty { get; set; }
    public ApplicationUserDTO Tenant { get; set; }
}

public class PurchaseRequestCreate
{
    public int RealtyId { get; set; }
}

File: ZoneSentry/ClientApp/src/api/models/PurchaseRequest.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ApplicationUser } from './ApplicationUser';
import type { Realty } from './Realty';

export type PurchaseRequest = {
    id?: number;
    sellCost?: number;
    realty?: Realty;
    newOwner?: ApplicationUser;
};